edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3511,4 +3511,35 @@ public Object doInHibernate(Session session) throws HibernateException {
 		}
 
 	}
+
+
+	/**
+	 * Return the grade changes made since a given time
+	 * @param assignmentIds	ids of assignments to check
+	 * @param since	timestamp from which to check for changes
+	 * @return set of changes made
+	 */
+	@Override
+	public List<GradingEvent> getGradingEvents(final List<Long> assignmentIds, final Date since) {
+		List<GradingEvent> rval = new ArrayList<>();
+
+		if (since == null) {
+			log.debug("No `since` timestamp was specified.  Returning null");
+			return null;
+		}
+
+		HibernateCallback hc = new HibernateCallback() {
+			@Override
+			public Object doInHibernate(Session session) throws HibernateException, SQLException {
+				Query q = session.createQuery("from GradingEvent as ge where ge.dateGraded >= :since" +
```

Comment:
Will likely need an index on these two columns for an efficient lookup, which may be a combination of both fields. But need to profile with lots of rows. Are you able to load up the db then do an explain plan on this?
````
SELECT
* 
FROM gb_grading_event_t ge
WHERE ge.date_graded >= :since
AND ge.gradable_object_id in (:assignment_ids);
````

Alternative, grab the actual generated SQL by turinng on the hibernate show sql feature in sakai.properties, just wondering if that gradeableObject.id does an implicit join.